.github/workflows/console-web-docker-build.yml

```diff
@@ -73,3 +62,69 @@ jobs:
       - name: Build the Docker image
         if: steps.validation.outputs.enabled == 'true' && github.event_name == 'pull_request'
         run: packages/docker/script/dc.sh build deploy-web --build-arg DEPLOYMENT_ENV=production
+
+      - name: Export docker image
+        if: steps.validation.outputs.enabled == 'true' && github.event_name == 'pull_request'
+        run: docker save console-deploy-web:latest | gzip > console-deploy-web.tar.gz
+
+      - name: Cache docker image
+        if: steps.validation.outputs.enabled == 'true' && github.event_name == 'pull_request'
+        uses: actions/cache@v4
+        with:
+          path: console-deploy-web.tar.gz
+          key: ${{ env.DOCKER_IMAGE_CACHE_KEY }}
+
+  e2e-tests:
+    runs-on: ubuntu-latest
+    needs: build-deploy-web
+    if: needs.build-deploy-web.outputs.skipped != 'true' && github.event_name == 'pull_request'
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-app-deps
+        with:
+          app: deploy-web
+      - name: Install Playwright Dependencies
+        run: npx playwright install-deps chromium
+      - name: Cache Playwright browsers
+        id: cache-playwright
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/ms-playwright
+          key: deploy-web-playwright-${{ runner.os }}-${{ hashFiles('**/package-lock.json') }}
+      - name: Install Playwright Browsers
+        if: steps.cache-playwright.outputs.cache-hit != 'true'
+        run: npx playwright install chromium
+      - name: Restore Docker image from cache
+        uses: actions/cache@v4
+        with:
+          path: console-deploy-web.tar.gz
+          key: ${{ needs.build-deploy-web.outputs.deploy-web-image }}
+      - name: Load Docker image from cache
+        run: docker load -i console-deploy-web.tar.gz
+      - name: Run e2e tests
+        env:
+          TEST_WALLET_MNEMONIC: ${{ secrets.CONSOLE_WEB_E2E_TEST_WALLET_MNEMONIC }}
+        run: |
+          docker run -d \
```

Comment:
non-blockin. having an env file would make it more reliable. we'd use same values for running tests locally and in ci.